examples/chip-tool/commands/common/Commands.cpp

```diff
@@ -328,9 +329,18 @@ CHIP_ERROR Commands::RunCommand(int argc, char ** argv, bool interactive,
 
 #endif // CONFIG_USE_LOCAL_STORAGE
 
+    mActiveCommand = command;
     return command->Run();
 }
 
+void Commands::SetCommandExitStatus(CHIP_ERROR status)
+{
+    if (mActiveCommand != nullptr)
+    {
+        return mActiveCommand->SetCommandExitStatus(status);
```

Comment:
It's really not clear to me that this handles our two event loop models correctly.  `#if !CONFIG_USE_SEPARATE_EVENTLOOP` then this is safe to call, presumably... though nothing guarantees our signal handler is in fact called on the Matter (main) thread, really.

But if we have a separate event loop, then the command exit status should only be touched on the Matter queue, right?  And that's running on some non-main thread.

Reply:
If the purpose is to revoke the inner command, is it still a problem? Or could you let me know what problem may it be? StartWaiting() and StopWaiting() have 2 implementations with/without CONFIG_USE_SEPARATE_EVENTLOOP that it seems can handle it, isn't it correct?

Reply:
This is a problem independent of what the desired behavior is.   You need to do SetCommandExitStatus on the thread it expects to happen on, if you want to avoid having thread races....

Reply:
If the expect behavior is to cancel the running command, could you let me know what function can I reference to add SetCommandExitStatus() on other threads like you mentioned? 
Take the pairing command as the example. If issuing "pairing ble-wifi" and wait until time-out, PairingCommand::OnPairingComplete() is called with Timeout. It is also just to call SetCommandExitStatus(err); Won't it have the thread racing problem? Do I miss anything?

Reply:
SetCommandExitStatus is called on the thread where the command is running, no?  For PairingCommand::OnPairingComplete, for example, that's on whatever thread the Matter event queue is running on.  It changes the status and then calls CHIPCommand::StopWaiting, which handles the thread synchronization with other threads.  But first it sets the exit status, and that part should only be touched on the thread where the command lives.